src/providers/TokenListProvider.tsx

```diff
@@ -0,0 +1,86 @@
+import { createContext, ReactNode, useCallback, useEffect, useState } from 'react';
+import { erc20Abi } from 'viem';
+import { useAccount } from 'wagmi';
+import { TOKEN_LIST } from '~/data/tokens';
+import { useCustomClient } from '~/hooks/useCustomClient';
+import { TokenData } from '~/types';
+
+type TokenBalance = {
+  tokenData: TokenData;
+  balance: string;
+};
+
+type TokenMap = {
+  [k: string]: TokenBalance;
+};
+
+type ContextType = TokenBalance[];
+interface TokenProps {
+  children: ReactNode;
+}
+
+export const TokenListContext = createContext({} as ContextType);
+
+export const TokenListProvider = ({ children }: TokenProps) => {
+  const [tokenHashMap, setTokenHashMap] = useState<TokenMap | null>(null);
+
+  const { address, chain } = useAccount();
+  const customClient = useCustomClient();
+
+  const loadTokensBalance = useCallback(
+    async (tokens: TokenData[]) => {
+      try {
+        if (!address) throw new Error('Address is required');
+
+        const multicallContractsArg = tokens.map((t) => {
+          return {
+            address: t.address,
+            abi: erc20Abi,
+            functionName: 'balanceOf',
+            args: [address],
+          };
+        });
+
+        const balance = await customClient.publicClient.multicall({
+          contracts: [...multicallContractsArg],
+        });
+
+        const formattedBalance = balance.reduce<TokenBalance[]>((acc, current, idx) => {
+          const balance = current.result?.toString() ?? '0';
+          const newVal = {
+            tokenData: tokens[idx],
+            balance: balance,
```

Comment:
Can be replaced:
{ 
tokenData: tokens[idx],
balance,
}

Reply:
https://github.com/0xShishigami/feo-web3-react-boilerplate/pull/1/commits/077c00abfe1ff2c00f95f1fea9553d7d56fb7f81